lib/elixir/lib/kernel.ex

```diff
@@ -3904,35 +4027,48 @@ defmodule Kernel do
                     Map.replace!(map, key, val)
                   end)
                 end
+
               _ ->
                 def __struct__(kv) do
                   {map, keys} =
                     Enum.reduce(kv, {@struct, @enforce_keys}, fn {key, val}, {map, keys} ->
                       {Map.replace!(map, key, val), List.delete(keys, key)}
                     end)
+
                   case keys do
-                    [] -> map
-                    _  -> raise ArgumentError, "the following keys must also be given when building " <>
-                        "struct #{inspect __MODULE__}: #{inspect keys}"
+                    [] ->
+                      map
+
+                    _ ->
+                      raise ArgumentError,
+                            "the following keys must also be given when building " <>
+                              "struct #{inspect(__MODULE__)}: #{inspect(keys)}"
                   end
                 end
             end
           end
+
         false ->
           quote do
             _ = @enforce_keys
+
             def __struct__(kv) do
-              :lists.foldl(fn {key, val}, acc ->
-                Map.replace!(acc, key, val)
-              end, @struct, kv)
+              :lists.foldl(
+                fn {key, val}, acc ->
```

Comment:
I believe you could try to have all of this on a single line.  If it doesn't git, then extract the functino to a variable.